integrations/nifi/nifi-nar-bundles/nifi-spark-bundle/nifi-spark-processors/src/main/java/com/thinkbiganalytics/nifi/v2/spark/ExecuteSparkJob.java

```diff
@@ -398,103 +406,50 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
-                args = appArgs.split(",");
-            }
 
-            final List<String> extraJarPaths = new ArrayList<>();
-            if (!StringUtils.isEmpty(extraJars)) {
-                extraJarPaths.addAll(Arrays.asList(extraJars.split(",")));
-            } else {
-                getLog().info("No extra jars to be added to class path");
-            }
+            final List<String> extraJarPaths = getExtraJarPaths(extraJars);
 
             // If all 3 fields are filled out then assume kerberos is enabled, and user should be authenticated
-            boolean authenticateUser = false;
-            if (!StringUtils.isEmpty(principal) && !StringUtils.isEmpty(keyTab) && !StringUtils.isEmpty(hadoopConfigurationResources)) {
-                authenticateUser = true;
-            }
-
-            if (authenticateUser) {
-                ApplySecurityPolicy applySecurityObject = new ApplySecurityPolicy();
-                Configuration configuration;
-                try {
-                    getLog().info("Getting Hadoop configuration from " + hadoopConfigurationResources);
-                    configuration = ApplySecurityPolicy.getConfigurationFromResources(hadoopConfigurationResources);
-
-                    if (SecurityUtil.isSecurityEnabled(configuration)) {
-                        getLog().info("Security is enabled");
-
-                        if (principal.equals("") && keyTab.equals("")) {
-                            getLog().error("Kerberos Principal and Kerberos KeyTab information missing in Kerboeros enabled cluster. {} ", new Object[]{flowFile});
-                            session.transfer(flowFile, REL_FAILURE);
-                            return;
-                        }
-
-                        try {
-                            getLog().info("User authentication initiated");
-
-                            boolean authenticationStatus = applySecurityObject.validateUserWithKerberos(logger, hadoopConfigurationResources, principal, keyTab);
-                            if (authenticationStatus) {
-                                getLog().info("User authenticated successfully.");
-                            } else {
-                                getLog().error("User authentication failed.  {} ", new Object[]{flowFile});
-                                session.transfer(flowFile, REL_FAILURE);
-                                return;
-                            }
-
-                        } catch (Exception unknownException) {
-                            getLog().error("Unknown exception occurred while validating user : {}.  {} ", new Object[]{unknownException.getMessage(), flowFile});
-                            session.transfer(flowFile, REL_FAILURE);
-                            return;
-                        }
+            boolean isAuthenticated = !StringUtils.isEmpty(principal) && !StringUtils.isEmpty(keyTab) && !StringUtils.isEmpty(hadoopConfigurationResources);
+            try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)